tests/mnpe_test.py

```diff
@@ -0,0 +1,150 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+import pytest
+import torch
+from torch import Tensor
+from torch.distributions import Bernoulli
+
+from sbi.inference import MNPE
+from sbi.inference.posteriors.rejection_posterior import RejectionPosterior
+from sbi.inference.posteriors.vi_posterior import VIPosterior
+from sbi.neural_nets import posterior_nn
+from sbi.neural_nets.embedding_nets import FCEmbedding
+from sbi.utils import BoxUniform
+from sbi.utils.torchutils import process_device
+from sbi.utils.user_input_checks_utils import MultipleIndependent
+
+
+# toy simulator for continuous data with mixed parameters
+def mixed_param_simulator(theta: Tensor) -> Tensor:
+    """Simulator for continuous data with mixed parameters.
+
+    Args:
+        theta: Parameters with mixed types - continuous and discrete.
+    Returns:
+        x: Continuous observation.
+    """
+    device = theta.device
+
+    # Extract parameters
+    a, b = theta[:, 0], theta[:, 1]
+    noise = 0.05 * torch.randn(a.shape, device=device).reshape(-1, 1)
+    return (a + 2 * b).reshape(-1, 1) + noise
```

Comment:
can we come up with a simulator for which we have ground truth posterior to make a test on the accuracy? 

e.g., a mixture of Gaussians where the discrete params just selects the component: 

z ~ Categorical
x ~ N(mu_z, 1)

and Gaussian priors on mu_z with different prior means, e.g., -1 and 1. 

then for a fixed z, the posterior will just be a Gaussian composed from the likelihood and prior Gaussian just like in our standard linear Gaussian example.